### PR TITLE
Fix the Check Test Logic and implement AllAssignedTests

### PR DESCRIPTION
**Pull request by vlad12345123**

### CODE REVIEW

File: exalearn-api/ExaLearn.Bl/DTO/PassedTestForCheckDTO.cs [hunk 1]
```diff
@@ -1,18 +1,9 @@
...
-
-        public LevelType LevelType { get; set; }             
-
-        public StatusType Status { get; set; }
-
-       // public TestDTO UserTest { get; set; }
+        public QuestionDTO[] Questions { get; set; }
 
-        public UserAnswerDTO[] UserAnswers { get; set; }
+        public UserAnswerQuestionDTO[] UserAnswers { get; set; }
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Bl/DTO/PassedTestForCheckDTO.cs]: answer or question?

File: exalearn-api/ExaLearn.Bl/Services/UserService.cs [hunk 2]
```diff
@@ -77,5 +77,11 @@ public async Task<List<PassedTestDTO>> MyTestHistoryAsync(int id)
 
             return _mapper.Map<List<PassedTestDTO>>(passedTests);
         }
+
+        public async Task<List<AssignedTestDTO>> GetAllAssignedTests() //dont remember which fields the fronts have on their page
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.Bl/Services/UserService.cs:81]: forget or remember?

File: exalearn-api/ExaLearn.WebApi/Controllers/UserController.cs [hunk 3]
```diff
@@ -75,5 +75,11 @@ public async Task<IActionResult> MyTestHistoryAsync(int id)
         {
             return Ok(await _userService.MyTestHistoryAsync(id));
         }
+
+        [HttpGet("{id}/allAssignedTests")]
```
Comment (ExadelSandbox) [inline exalearn-api/ExaLearn.WebApi/Controllers/UserController.cs]: why do you need here id? how does it treated?

**Review by ExadelSandbox — APPROVED** [review]
after corrections it is possible to merge

File: exalearn-api/ExaLearn.Bl/Services/UserService.cs (hunk 2 above)
Comment (vlad12345123) [reply exalearn-api/ExaLearn.Bl/Services/UserService.cs:81]: forget:D

**Review by artemgael96 — APPROVED** [review]

**Pull request merged**